### PR TITLE
Feat/email recipients

### PR DESCRIPTION
**Pull request by samsmith89**
This PR covers the following: 
- Added option for users to select time of when the giftee should be notified
- Giftee receives access, emails, and prompt to be added to mail list at specified time
- Adjusted Cron schedule to check for giftees every 30 minutes and send every 5 minutes
- Saved email sending as Unix timestamps instead of dates for more reliable parsing

### CODE REVIEW

File: includes/class-eddcoursegiftingcronmod.php [hunk 1]
```diff
@@ -56,14 +56,14 @@ public static function get_instance() {
...
 
 	public function wdm_cron_cchedules( $schedules ) {
-		if ( ! isset( $schedules['wdm_daily_emails_gift_reminder'] ) ) {
-			$schedules['wdm_daily_emails_gift_reminder'] = array(
-				'interval' => 86400,
-				'display' => __( 'Once a day', 'learndash-gift-edd' ),
+		if ( ! isset( $schedules['wdm_emails_gift_reminder'] ) ) {
+			$schedules['wdm_emails_gift_reminder'] = array(
+				'interval' => 1800,
+				'display' => __( 'After 30 min', 'learndash-gift-edd' ),
```
Comment (landbryo) [inline includes/class-eddcoursegiftingcronmod.php]: I recommend adding a name that represents the interval and not the function. So, maybe just remove "After" so it is "30 minutes".

File: includes/class-learndasheddgift.php [hunk 2]
```diff
@@ -238,11 +238,7 @@ public function plugin_activation_handler() {
 		if ( ! wp_next_scheduled( 'emails_gift_reminder' ) ) {
 			$date         = gmdate( 'Y-m-d' );
 			$current_time = strtotime( $date . ' 07:00:00' );
-			if ( $current_time > time() ) {
-				wp_schedule_event( $current_time, 'wdm_daily_emails_gift_reminder', 'emails_gift_reminder' );
-			} else {
-				wp_schedule_event( strtotime( '+1 day', $current_time ), 'wdm_daily_emails_gift_reminder', 'emails_gift_reminder' );
-			}
+			wp_schedule_event( $current_time, 'wdm_emails_gift_reminder', 'emails_gift_reminder' );
```
Comment (landbryo) [inline includes/class-learndasheddgift.php]: I know this is how it was before, but any reason it is getting the current time like this? Seems like getting the actual current time would be better.

File: public/class-eddcoursegiftingcheckoutmod.php [hunk 3]
```diff
@@ -118,7 +118,7 @@ public function save_purchaser_gift_data( $payment_id ) {
 			update_post_meta( $payment_id, 'edd_ld_gift_last_name', $last_name );
 			if ( isset( $_POST['send_later_status'] ) && $_POST['send_later_status'] ) {
 				update_post_meta( $payment_id, 'send_later_status', true );
-				update_post_meta( $payment_id, 'edd_ld_gift_date', $_POST['edd_ld_gift_date'] );
+				update_post_meta( $payment_id, 'edd_ld_gift_date', $_POST['edd_ld_gift_timestamp']);
```
Comment (landbryo) [inline public/class-eddcoursegiftingcheckoutmod.php]: Should be sanitized.

**Review by landbryo — CHANGES_REQUESTED** [review]
Looks good. Minor adjustments suggested.

File: includes/class-learndasheddgift.php (hunk 2 above)
Comment (samsmith89) [reply includes/class-learndasheddgift.php]: None that I can see. time() is used in the docs and that's good enough for me.

File: includes/class-eddcoursegiftingcronmod.php (hunk 1 above)
Comment (samsmith89) [reply includes/class-eddcoursegiftingcronmod.php]: Changed

File: public/class-eddcoursegiftingcheckoutmod.php (hunk 3 above)
Comment (samsmith89) [reply public/class-eddcoursegiftingcheckoutmod.php]: Agreed. Fixed

**Pull request merged**